libs/blocks/merch/merch.js

```diff
@@ -122,6 +122,34 @@ const LOADING_ENTITLEMENTS = 'loading-entitlements';
 let log;
 let upgradeOffer = null;
 
+/**
+ * Given a url, calculates the hostname of MAS platform.
+ * Supports, www prod, stage, local and feature branches.
+ * @param {string} hostname
+ * @param {string} maslibs
+ * @returns base url for mas platform
+ */
+export function getMasBase(hostname, maslibs) {
+  let { baseUrl } = getMasBase;
```

Comment:
We used this pattern in the other functions in merch block. Thus, we can cache the calculated value, and erase it when needed (mostly in tests). Using a module scope variable makes testing harder. You need to provide a utility method to reset the module scope variables.

Reply:
But still, on the first pass the baseUrl value would be empty and I don't see a reason why you would need to call the method multiple times.

Reply:
@narcis-radu I removed the caching part as it is not going to be called to many times but several times definitely.
later after commerce.js, we will start adding maslibs feature to merch-card, merch-card-collection and other merch blocks as well.
merch being auto block it will be the first one to call this method and have the base url initialised, then the other blocks can use the value.
@3ch023 I'm tempted to add merch-card in this PR, thus we can give a better shape to the code, wdyt? cc: @npeltier

Reply:
I am still not happy with the `baseUrl` approach since it doesn't follow the pattern used in other Milo blocks, but I don't think this is a blocker or reason to request changes. Thanks for addressing the important changes 👍

Reply:
Actually,   something like `getMasLibs` `setMasLibs` would be better. I'll give it a try tomorrow my morning.